src/object/srv_obj.c

```diff
@@ -533,7 +533,19 @@ obj_bulk_transfer(crt_rpc_t *rpc, crt_bulk_op_t bulk_op, bool bulk_bind,
 	for (i = 0; i < sgl_nr; i++) {
 		d_sg_list_t	*sgl, tmp_sgl;
 
-		if (remote_bulks[i] == NULL)
+		/* Need to consider the akey skip case as client-side bulk array with one bulk for
+		 * every akey, but some akeys possibly be skipped in obj_get_iods_offs (number of
+		 * akeys possibly reduced). So here need to find the corresponding bulk handle
+		 * for non-skipped akey. The akey skip case only possible for EC object and targeted
+		 * for multiple data shards.
+		 * For RPC inline (non-bulk) case (bio_iod_copy) need not consider the akey skip
+		 * because we always create bulk handle if EC object IO targeted for multiple
+		 * data shards.
+		 */
+		if (skips != NULL && isset(skips, i))
+			skip_nr++;
```

Comment:
Some confused, why not directly "continue" for the skipped case? The "sgl_nr" is equal to the length of remote_bulks array or not?

Reply:
cannot continue here, the sgl_nr (returned by obj_get_iods_offs) possible not equal to the length of remote_bulks (client created and in RPC).
this patch with a few issues and cannot pass one test, still need a few debug

Reply:
Then need to guarantee that the "i +skip_nr" will exceeds the remote_bulks array boundary.

Reply:
sure, will add assertion check, thx